packages/components/switch/src/less/switch.less

```diff
@@ -48,22 +54,23 @@
 
     &:checked {
       ~ .oui-switch__slider {
+        border-color: @oui-switch-border-color_checked;
         background-color: @oui-switch-background-color_checked;
       }
 
       ~ .oui-switch__slider::before {
-        transform: translateX(@oui-switch-width - @oui-switch-height);
+        content: @oui-switch-cursor-icon-index_checked;
+        color: @oui-switch-cursor-icon-color_checked;
+        box-shadow: @oui-switch-cursor-box-shadow_checked;
+        background-color: @oui-switch-cursor-background-color_checked;
+        transform: translateX(@oui-switch-width - @oui-switch-cursor-size - (@oui-switch-cursor-position-x * 2) - (@oui-switch-border-width * 2));
       }
     }
 
     &:disabled {
       ~ .oui-switch__slider {
-        background-color: @oui-switch-background-color_disabled;
-        cursor: default;
-
-        &::before {
-          background-color: @oui-switch-cursor-background-color_disabled;
-        }
+        opacity: 0.5;
```

Comment:
How do you choose when to use a `variable` and when not to do so ?

Reply:
When it's a value that could be easily changed and centralized (eg. color, size, spacing, etc.).
Here for `opacity`, it's a bit like `display` and should not change when there's an update of the tokens.
But maybe we could centralized that (in a mixin), since it's behaviour shared by all the components: a disabled component keeps its default state, with an opacity of `0.5`.

Reply:
I like this idea :)